coretest/coretest.go

```diff
@@ -0,0 +1,309 @@
+package coretest
```

Comment:
I think this is a good idea.

I've seen it referred to more often than not as a `fake` package, more of a FYI, no strong preference.